WordPress/Classes/ViewRelated/Blog/BlogListViewController.m

```diff
@@ -210,12 +211,37 @@ - (void)updateSearchVisibility
 
 - (void)maybeShowNUX
 {
-    if (self.dataSource.allBlogsCount > 0) {
+    NSInteger blogCount = self.dataSource.allBlogsCount;
+    BOOL isLoggedIn = AccountHelper.isLoggedIn;
+
+    if (blogCount > 0 && !isLoggedIn) {
         return;
     }
+    
     if (![self defaultWordPressComAccount]) {
         [[WordPressAppDelegate shared] showWelcomeScreenIfNeededAnimated:YES];
+        return;
+    }
+
+    BOOL shouldDisplayPSI = [PostSignUpInterstitialViewController shouldDisplayWithNumberOfBlogs:blogCount];
+
+    if (!shouldDisplayPSI) {
+        return;
     }
+
+    [self showPostSignUpInterstitial];
+}
+
+#warning TODO: Change Name
```

Comment:
Shall we change the name and remove the `#warning`? :P

Reply:
👍I've just removed the warning, looking at the other code in the class using the word "show" is the correct terminology